Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -7119,8 +7119,11 @@ bool HTMLMediaElement::videoUsesElementFullscreen() const
 {
 #if ENABLE(FULLSCREEN_API) && ENABLE(VIDEO_USES_ELEMENT_FULLSCREEN)
 #if ENABLE(LINEAR_MEDIA_PLAYER)
-    if (document().settings().linearMediaPlayerEnabled())
-        return false;
+    if (document().settings().linearMediaPlayerEnabled()) {
+        RefPtr player = m_player;
+        if (player && player->supportsLinearMediaPlayer())
```

Comment:
could be a one liner `if (RefPtr player = m_player; player && player->supportsLinearMediaPlayer());`